docs/notebooks/asynchronous_bayesian_optimization.pct.py

```diff
@@ -0,0 +1,373 @@
+# 
+# The overall setup is illustrated in this diagram:
+# ![multiprocessing setup](figures/async_bo.png)
+
+# %%
+# Necessary multiprocessing primitives
+from multiprocessing import Manager, Process
+
+# %% [markdown]
+# We now define several common functions to implement the described setup. First we define a worker function that will be running a single observation in a separate process. Worker takes both queues as an input, reads next point from the points queue, makes an observation, and inserts observed data into the observations queue.
+
+# %%
+
+def observer_proc(points_queue, observations_queue):
+    pid = os.getpid()
+
+    while True:
+        point_to_observe = points_queue.get()
+        if point_to_observe is None:
+            return
+
+        print(f"Process {pid}: Observer : observing data at point {point_to_observe}", flush=True)
+        new_observation = objective(point_to_observe, sleep=enable_sleep_delays)
+        new_data = (point_to_observe, new_observation)
+
+        print(f"Process {pid}: Observer : observed data {new_data}", flush=True)
+
+        observations_queue.put(new_data)
+
+# %% [markdown]
+# Next we define two helper functions, one is to create a certain number of worker processes, and another is to terminate them once we are done.
+
+# %%
+
+def create_worker_processes(n_workers, points_queue, obseverations_queue):
+    observer_processes = []
+    for i in range(n_workers):
+        worker_proc = Process(target=observer_proc, args=(points_queue, obseverations_queue))
+        worker_proc.daemon = True
+        worker_proc.start()
+
+        observer_processes.append(worker_proc)
+
+    return observer_processes
+
+def terminate_processes(processes):
+    for prc in processes:
+        prc.terminate()
+        prc.join()
+        prc.close()
+
+# %% [markdown]
+# Finally we set some common parameters. See comments below for explanation of what each one means.
+# %%
+# Number of worker processes to run simultaneously
+# Setting this to 1 will turn both setups into non-batch sequential optimization
+num_workers = 3
+# Number of iterations to run the sycnhronous scenario for
+num_iterations = 10
+# Number of observations to collect in asynchronous scenario
+num_observations = num_workers * num_iterations
+# Set this flag to False to disable sleep delays in case you want the notebook to execute quickly
+enable_sleep_delays = True
```

Comment:
Is there a way to set this to be False, when we run our notebook tests? i.e checking an environment variable? @hstojic @uri-granta ?

Reply:
We could do something similar in most of our notebooks, as they are currently super slow when testing the package.

Reply:
Was pondering the same question waiting for the builds to run. Will research

Reply:
So, we can use env variable, or even [papermill](https://github.com/yaananth/papermill) which allows to parametrize notebooks. But we will have to add something ugly in the tutorial, along these lines:
```
if "async-bo-enable-sleep" in os.environ:
    # long comment explaining the reader why this is here, about integ tests and what not
    enable_sleep_delays = os.getenv("async-bo-enable-sleep")
else:
    enable_sleep_delays = True
```

i find this ugly in a tutorial. Shall we just downscale the delay?

Reply:
if we name the variable "running_CI" or something similar, then we dont need a comment to explain whats going on.

Reply:
#371